pkg/resources/aws.go

```diff
@@ -2054,3 +2082,34 @@ func FindELBName(tags []*elb.Tag) string {
 	}
 	return ""
 }
+
+// HasSharedTag looks for the shared tag indicating that the cluster does not own the resource
+func HasSharedTag(description string, tags []*ec2.Tag, clusterName string) bool {
+	tagKey := "kubernetes.io/cluster/" + clusterName
+
+	var found *ec2.Tag
+	for _, tag := range tags {
+		if aws.StringValue(tag.Key) != tagKey {
+			continue
+		}
+
+		found = tag
+	}
+
+	if found == nil {
+		glog.Warningf("(new) cluster tag not found on %s", description)
+		return false
+	}
+
+	tagValue := aws.StringValue(found.Value)
+	switch tagValue {
+	case "owned":
+		return false
+	case "shared":
+		return true
+
+	default:
+		glog.Warningf("unknown cluster tag on %s: %q=%q", description, tagKey, tagValue)
+		return false
```

Comment:
I think this should actually default to `true`, as owned resources would be tagged properly upon creation, but shared (e.g. pre-existing would not be).

Reply:
Created #3119 by the way